Lifted Product construction of CoPrime Bivariate Bicycle codes

Fe-r-oz opened the pull request:
This PR aims to introduce an easy construction methods for coprime bivaraite bicycle codes using lifted product `LPCode`, and Oscar `SubPcGroup`.  The code_n and code_k parameters match exactly from the table 2 of https://arxiv.org/pdf/2408.10001v1.


- [x] The code is properly formatted and commented.
- [x] Substantial new functionality is documented within the docs.
- [x] All new functionality is tested.
- [x] All of the automated tests on github pass.

Fe-r-oz commented:
I think the PR is ready for review. Thank you!

Fe-r-oz commented:
Since #394 is fixed, I can now add the tests that reproduce all the results from Table 2 for these codes. I waited because otherwise, it would have caused CI failures.

Krastanov commented on ext/QuantumCliffordHeckeExt/lifted_product.jl:
````diff
@@ -70,6 +70,35 @@ julia> code_n(c2), code_k(c2)
 - When the two matrices are adjoint to each other, the code is called a bicycle code [`bicycle_codes`](@ref).
 
+
+# Examples
+
+The coprime bivariate bicycle (BB) codes are defined by two polynomials `𝑎(𝑥,𝑦)` and `𝑏(𝑥,𝑦)`,
+where `𝑙` and `𝑚` are coprime, and can be expressed as univariate polynomials `𝑎(𝜋)` and `𝑏(𝜋)`,
+with generator `𝜋 = 𝑥𝑦`. They can be viewed as a special case of Lifted Product construction
+based on abelian group `ℤₗ x ℤₘ` where `ℤⱼ` cyclic group of order `j`.
+
+[108, 12, 6]] coprime-bivariate bicycle (BB) code from Table 2 of [wang2024coprime](@cite).
+
+```jldoctest
+julia> import Hecke: group_algebra, GF, abelian_group, gens;
+
+julia> l=2; m=27;
+
+julia> GA = group_algebra(GF(2), abelian_group([l*m]));
+
+julia> 𝜋 = gens(GA)[1];
+
+julia> A = reshape([𝜋^2 + 𝜋^5 + 𝜋^44], (1,1));
+
+julia> B = reshape([𝜋^8 + 𝜋^14 + 𝜋^47], (1,1));
+
+julia> c = LPCode(A, B);
+
+julia> code_n(c), code_k(c)
+(108, 12)
+```
````
This seems like an example of a `two_block_group_algebra_codes` so it should probably go in its docstring, not here, next to the other example you have already given (that I merged earlier today, so presumably this also needs a merge with master to avoid a merge conflict).

Krastanov commented on test/test_ecc_coprime_bivaraite_bicycle.jl:
```diff
@@ -0,0 +1,59 @@
+@testitem "ECC coprime Bivaraite Bicycle" begin
```
very useful set of tests! It should probably directly use `two_block_group_algebra_code` instead of the more round-about construction with LPCode

Krastanov reviewed (changes requested):
This is wonderful, thank you! I think it would make sense to put the docstring in a different location though (see comments below)

Fe-r-oz commented:
Thank you for your suggestions! Incorporated all of your comments and added coprimeBB codes in `test_ecc_base `to test their properties as well!

This PR is ready for review, Thank you!
